include/ceed/jit-source/cuda/cuda-shared-basis-tensor-at-points-templates.h

```diff
@@ -0,0 +1,568 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and other CEED contributors.
```

Comment:
Hmm, it feels weird that the only difference between the HIP and CUDA files is the type of the `SharedData_*` parameter. I suppose for clarity (and consistency), it makes sense to fully separate HIP and CUDA backends, but for ease of maintenance, is it possible we could do something like, for example:
```c
template <typename SharedData_t, int NUM_COMP, int NUM_POINTS, int Q_1D>
inline __device__ void InterpAtPoints1d(SharedData_t &data, const CeedInt p, const CeedScalar *__restrict__ r_C, const CeedScalar *r_X,
                                        CeedScalar *__restrict__ r_V) {
```

This isn't necessarily a suggestion for this PR, just a thought about simplifications we could make in the future to reduce duplication

Reply:
There are some other small differences too. I'm a little hesitant to make more template parameters right now given some of the frustration PETSc has felt doing something similar.